napari/_qt/_qapp_model/_tests/test_dock_widget_toggler.py

```diff
@@ -0,0 +1,25 @@
+from napari._app_model._app import get_app
+from napari._qt._qapp_model.qactions._window import ToggleDockWidgetAction
+from napari._qt.qt_main_window import Window
+
+
+def test_dock_widget_toggler(make_napari_viewer):
+    """Tests `ToggleDockWidgetAction` toggling works."""
+    viewer = make_napari_viewer(show=True)
+    action = ToggleDockWidgetAction(
+        id='some.command.id',
+        title='Toggle Dock Widget',
+        dock_widget='dockConsole',
+    )
+
+    app = get_app()
+    app.register_action(action)
+
+    with app.injection_store.register(
+        providers={Window: lambda: viewer.window}
```

Comment:
I would define a named function for the `window` provider here. Lambdas holding references to qt stuff has bit us before.

Reply:
This injection should not be required as test executed standalone works. So some cleaning after some of the previous tests broke it and such cleaning should be fixed.

Reply:
@DragaDoncila do you recall why we thought the injection was needed here? Note in the similar test `test_viewer_toggler` we also inject `viewer` even though we run `viewer = ViewerModel()` above?

https://github.com/napari/napari/blob/af7d46918a67084e6c8e927251936e417ee6d5cc/napari/_app_model/_tests/test_viewer_toggler.py#L18